src/pages/Auth/Auth.tsx

```diff
@@ -0,0 +1,40 @@
+import { useActionData, Form, useNavigate } from "react-router-dom";
+import { COMMON_ROUTES_NAMES } from "@/routing";
+import { AuthActionData } from '@/actions'
+import styles from './Auth.module.scss';
+import { useAuth } from "@/firebase";
+import { AuthStatus } from "@/actions/actionsAuth";
+import { ROLES } from "@/types";
+import { useEffect } from "react";
+
+const Auth = () => {
+	const actionData = useActionData() as AuthActionData | undefined;
+	const { role } = useAuth();
+	const navigate = useNavigate();
+
+	useEffect(() => {
```

Comment:
I know it was my idea to put it like this, but I just got myself thinking that maybe you could just try to use [Navigate component](https://reactrouter.com/en/main/components/navigate) instead.

in case of using `Navigate` component you would simply need something like this in your JSX:
```
actionData?.status === AuthStatus.Success && Object.values(ROLES).includes(role!) && <Navigate to="/" replace={true}/>
```